pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -107,8 +107,7 @@ def advance(  # type: ignore[override]
             raise StopIteration
 
         if not data_fetcher.store_on_device:
-            with self.trainer.profiler.profile("evaluation_batch_to_device"):
-                batch = self.trainer.training_type_plugin.batch_to_device(batch, dataloader_idx=(dataloader_idx or 0))
```

Comment:
this is very interesting. now everything will show up as `batch_to_device`. this is positive for consistency with users since there's no "evaluation_batch_to_device" hook users implement. on the other hand, what happens if users implement these hooks differently based on the running stage the trainer is in?

we don't profile the `setup` or `teardown` hooks any differently, so i wonder if `batch_to_device` has just been an anomaly this whole time

Reply:
@ananthsub how can they implement these hooks differently based on the running stage the trainer? ttk passed in when create trainer which will be shared across stages right?

Reply:
@four4fish the training type plugin calls the lightning module hook. the lightning module hook can be overridden by users like this:

```
def transfer_batch_to_device(self, batch, device, dataloader_idx):
    if self.trainer.training:
        .....
   elif self.trainer.testing:
        ....
```

Reply:
In addition to checking the running stage, the performance could be different if each stage's data is very different. But I don't think it warrants a specific profile name

Reply:
this can (should) be mentioned in the changelog (and the ones below too)